tests/shadowing_parameters.rs

```diff
@@ -0,0 +1,25 @@
+use super::*;
+
+test! {
+  name:     parameter_may_shadow_variable,
+  justfile: "FOO := 'hello'\na FOO:\n echo {{FOO}}\n",
+  args:     ("a", "bar"),
+  stdout:   "bar\n",
+  stderr:   "echo bar\n",
+}
+
+test! {
+  name:     shadowing_parameters_do_not_change_environment,
+  justfile: "export FOO := 'hello'\na FOO:\n echo $FOO\n",
+  args:     ("a", "bar"),
+  stdout:   "hello\n",
```

Comment:
It's not immediately clear to me why passing "bar" on the command line doesn't override "hello" here, but it does in the other two cases.

Reply:
In this case, `$FOO` refers to an environment variable, which comes from the outer export. In the first test, `FOO` refers to the just variable, not the environment variable. In the last case, the just parameter `FOO` is exported with `$FOO`, so it now refers to the exported parameter.